common/unified/solver/chebyshev_kernels.cpp

```diff
@@ -0,0 +1,65 @@
+// SPDX-FileCopyrightText: 2017 - 2025 The Ginkgo authors
+//
+// SPDX-License-Identifier: BSD-3-Clause
+
+#include "core/solver/chebyshev_kernels.hpp"
+
+#include <ginkgo/core/matrix/dense.hpp>
+
+#include "common/unified/base/kernel_launch.hpp"
+
+
+namespace gko {
+namespace kernels {
+namespace GKO_DEVICE_NAMESPACE {
+namespace chebyshev {
+
+
+template <typename ValueType, typename ScalarType>
```

Comment:
Does this actually need a second precision, or can we always use the highest precision available? This is memory-bound after all.

Reply:
I have the same thought and thinking maybe we just use double or complex<double> directly?

Reply:
If there is a numerical benefit to using a higher-precision multiplication every time, I would go for it

Reply:
after rethink about it, I do not have a good way to use double or complex double without any information from the vector or matrix. Maybe still keep the same way as how we handle IR/richardson now. also I need the ScalarType, which is the ValueType from Chebyshev class

Reply:
currently keep the value type of class but change fosi, alpha, beta type to double/complex<double>
when dpcpp only support float not double, it will cast to float/complex<float> internally to keep the same interface among all backends